File: tests/js/fixtures.mjs
// Deterministic paper fixtures + a fetch() stub for testing js/core.js without
// touching the network or the real data/*.json files.

export const FIXTURE_PAPERS = [
  {
    // author search target #1: has "segmentation" in title+abstract
    title: 'Deep CNN Segmentation of MRI Scans',
    authors: 'Jane Smith, Bob Lee',
    url: 'https://example.org/paper-1',
    venue: 'MICCAI 2021',
    year: '2021',
    abstract: 'We propose a transformer-based approach for medical image segmentation.',
  },
  {
    // author search target #2: same author, no "segmentation"
    title: 'Calibration of Deep Neural Network Models',
    authors: 'Jane Smith',
    url: 'https://example.org/paper-2',
    venue: 'MICCAI 2019',
    year: '2019',
    abstract: 'We analyze confidence calibration techniques for classifiers.',
  },
  {
    // distractor: "smith" appears in title/abstract but NOT in authors
    title: "Extending Smith's Loss Function for Robust Training",
    authors: 'David Kim',
    url: 'https://example.org/paper-3',
    venue: 'AAAI 2021',
    year: '2021',
    abstract: 'We build on the smith regularizer to improve robustness.',
  },
  {
    // criss-cross with paper 1: "cnn" in abstract, "transformer" in title
    title: 'Transformer Networks for Visual Recognition',
    authors: 'Alice Doe',
    url: 'https://example.org/paper-4',
    venue: 'CVPR 2022',
    year: '2022',
    abstract: 'This work explores CNN alternatives for vision tasks.',
  },
  {
    // has "cnn" but NOT "transformer" - distinguishes AND from OR
    title: 'Federated Learning Survey with CNN Backbones',
    authors: 'Carol White, David Kim',
    url: 'https://example.org/paper-5',
    venue: 'ICML 2023',
    year: '2023',
    abstract: 'A survey of federated learning techniques.',
  },
  {
    // "graph" only in title, "molecule" only in abstract
    title: 'Graph Neural Networks for Chemistry',
    authors: 'Bob Lee, Carol White',
    url: 'https://example.org/paper-6',
    venue: 'NeurIPS 2020',
    year: '2020',
    abstract: 'We study molecule embeddings for property prediction tasks.',
  },
];

const FIXTURE_CONFIG = { conferences: [{ id: 'fixture' }] };

/**
 * Stubs globalThis.fetch so js/core.js's loadPapers() resolves the fixture
 * data above instead of hitting the network. Call once before any test that
 * exercises fetchResults()/loadPapers().
 */
export function installFetchMock() {
  globalThis.fetch = async (url) => {
    if (String(url).includes('config.json')) {
      return { ok: true, json: async () => FIXTURE_CONFIG };
    }
    if (String(url).includes('fixture.json')) {
      return { ok: true, json: async () => FIXTURE_PAPERS };
    }
    throw new Error(`Unexpected fetch() call in test: ${url}`);
  };
}

File: tests/python/fixtures.py
"""Deterministic paper fixtures for testing api.search.run_search().

Mirrors the shape api.search._build_index() produces internally: a list of
(lowered, normalized) tuples. Building these fixtures directly lets tests
bypass _build_index()'s live network fetchers entirely.
"""

# (title, authors, url, venue, year, abstract)
FIXTURE_RAW = [
    (
        "Deep CNN Segmentation of MRI Scans",
        "Jane Smith, Bob Lee",
        "https://example.org/paper-1",
        "MICCAI 2021",
        "2021",
        "We propose a transformer-based approach for medical image segmentation.",
    ),
    (
        "Calibration of Deep Neural Network Models",
        "Jane Smith",
        "https://example.org/paper-2",
        "MICCAI 2019",
        "2019",
        "We analyze confidence calibration techniques for classifiers.",
    ),
    (
        "Extending Smith's Loss Function for Robust Training",
        "David Kim",
        "https://example.org/paper-3",
        "AAAI 2021",
        "2021",
        "We build on the smith regularizer to improve robustness.",
    ),
    (
        "Transformer Networks for Visual Recognition",
        "Alice Doe",
        "https://example.org/paper-4",
        "CVPR 2022",
        "2022",
        "This work explores CNN alternatives for vision tasks.",
    ),
    (
        "Federated Learning Survey with CNN Backbones",
        "Carol White, David Kim",
        "https://example.org/paper-5",
        "ICML 2023",
        "2023",
        "A survey of federated learning techniques.",
    ),
    (
        "Graph Neural Networks for Chemistry",
        "Bob Lee, Carol White",
        "https://example.org/paper-6",
        "NeurIPS 2020",
        "2020",
        "We study molecule embeddings for property prediction tasks.",
    ),
]


def _entry(title, authors, url, venue, year, abstract):
    normalized = {
        "title": title,
        "authors": authors,
        "url": url,
        "venue": venue,
        "year": year,
        "abstract": abstract,
    }
    lowered = {
        "title": title.lower(),
        "authors": authors.lower(),
        "abstract": abstract.lower(),
        "venue": venue.lower(),
        "_all": f"{title} {authors} {abstract} {venue}".lower(),
    }
    return (lowered, normalized)


def build_fixture_index():
    return [_entry(*row) for row in FIXTURE_RAW]
